File: seminar-06/deep_first_search/java/dfs.java
import java.util.*;

public class dfs {

    // g: список смежности (0..n-1)
    public static void dfsRecursive(int v, List<List<Integer>> graph, boolean[] visited) {
        visited[v] = true;
        for (int to : graph.get(v)) {
            if (!visited[to]) {
                dfsRecursive(to, graph, visited);
            }
        }
    }
}

File: seminar-06/deep_first_search/cpp/dfs.cpp
#include <vector>
using namespace std;

void dfs_recursive(int v, const vector<vector<int>>& g, vector<char>& visited) {
    visited[v] = 1;
    for (int to : g[v]) {
        if (!visited[to]) dfs_recursive(to, g, visited);
    }
}